### PR TITLE
Add new keymap for Preonic rev3

### PR DESCRIPTION
**Pull request by guillermoap**
## Description
Created new keymap for Preonic rev3 with encoder support. As mentioned in the README this is highly curated to my liking but is a good example to use the ENCODER_MAP_ENABLE from the docs that I haven't seen in other keymaps.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/preonic/keymaps/guillermoap/keymap.c [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    } else if (layer_state_is(_RAISE)) {
+      if (clockwise) {
+        tap_code(KC_MS_WH_RIGHT);
+      } else {
+        tap_code(KC_MS_WH_LEFT);
+      }
+    }
+    */
+    return false;
+}
```
Comment (drashna) [inline keyboards/preonic/keymaps/guillermoap/keymap.c:188]:
```suggestion
```

File: keyboards/preonic/keymaps/guillermoap/keymap.c [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+/* Macros
+ * Ctrl + C
+ * Ctrl + Z
+ */
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case CTRL_C:
+            if (record->event.pressed) {
+                SEND_STRING(SS_LCTL("c"));
```
Comment (drashna) [inline keyboards/preonic/keymaps/guillermoap/keymap.c:199]: `LCTL(KC_C)` in your keymap will work for this as well (as will `C(KC_NC)`.

File: keyboards/preonic/keymaps/guillermoap/keymap.c [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+            }
+            return false;
+            break;
+        case CTRL_Z:
+            if (record->event.pressed) {
+                SEND_STRING(SS_LCTL("z"));
```
Comment (drashna) [inline keyboards/preonic/keymaps/guillermoap/keymap.c:205]: Same, here.

File: keyboards/preonic/keymaps/guillermoap/config.h [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+/* enable advanced MIDI features:
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 2
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/guillermoap/config.h:48]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by guillermoap** [comment]
@drashna @fauxpark Do I need to do anything else with this PR?

**Review by fauxpark — APPROVED** [review]

**Pull request merged**